skmob/measures/home_detection.py

```diff
@@ -0,0 +1,420 @@
+
+    The :math:`radius activity` can be be computed using a spatial join method to check which points are within around the specified radius or a .
+
+
+    Parameters
+    ----------
+    traj : TrajDataFrame
+        the trajectories of the individuals.
+
+    week_period : str, optional
+        the week period to consider. It can be 'WK' (week days), 'WE' (weekend days) or None (every day of the week). The default is None.
+
+    radius : int, optional
+        if specified, a spatial perimeter of a certain radius will be implemented around each location, the radius is expressed in kilometers. The default is None.
+    
+    mode : str, optional
+        the method used to compute the :math:`radius activity`. It can be 'sjoin' (spatial join) or 'distance_matrix' . The default is 'sjoin'.
+
+    show_progress : boolean, optional
+        if True, show a progress bar. The default is True.
+    
+    Returns
+    ----------
+    pandas DataFrame
+        the home location, as a :math:`(latitude, longitude)` pair, of the individuals.
+
+    Examples
+    --------
+    >>> import skmob
+    >>> from skmob.measures.home_detection import home_location_ma
+    >>> url = "https://snap.stanford.edu/data/loc-brightkite_totalCheckins.txt.gz"
+    >>> df = pd.read_csv(url, sep='\\t', header=0, nrows=100000, 
+                 names=['user', 'check-in_time', 'latitude', 'longitude', 'location id'])
+    >>> tdf = skmob.TrajDataFrame(df, latitude='latitude', longitude='longitude', datetime='check-in_time', user_id='user')
+    >>> hl = home_location_ma(tdf)
+    >>> print(hl.head())
+       uid        lat         lng
+    0    0  39.762146 -104.982480
+    1    1  37.630490 -122.411084
+    2    2  39.739154 -104.984703
+    3    3  37.748170 -122.459192
+    4    4  60.166970   24.944602
+    
+    References
+    ----------
+    .. [PFSCB2021] Pappalardo, Luca & Ferres, Leo & Sacasa, Manuel & Cattuto, Ciro & Bravo, Loreto. (2021). Evaluation of home detection algorithms on mobile phone data using individual-level ground truth. EPJ Data Science. 10. 29. 10.1140/epjds/s13688-021-00284-9. 
+    .. [CBTDHVSB2012] Csáji, B. C., Browet, A., Traag, V. A., Delvenne, J.-C., Huens, E., Van Dooren, P., Smoreda, Z. & Blondel, V. D. (2012) Exploring the Mobility of Mobile Phone Users. Physica A: Statistical Mechanics and its Applications 392(6), 1459-1473, https://www.sciencedirect.com/science/article/pii/S0378437112010059
+    .. [PSO2012] Phithakkitnukoon, S., Smoreda, Z. & Olivier, P. (2012) Socio-geography of human mobility: A study using longitudinal mobile phone data. PLOS ONE 7(6): e39253. https://doi.org/10.1371/journal.pone.0039253
+    """
+    return _uid_apply(_home_location_ma, traj, show_progress, week_period=week_period, radius=radius, mode=mode)
+
+
+def home_location_tc(traj, week_period=None, start_time='22:00', end_time='07:00', radius=None, mode='sjoin', show_progress=True):
```

Comment:
same comment as before. Let's use the name of the algorithm as a method name: time_constraint(..)